python/mxnet/gluon/contrib/estimator/estimator_model.py

```diff
@@ -0,0 +1,112 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=wildcard-import, unused-argument, too-many-ancestors
+"""Gluon EstimatorModels for Estimators"""
+
+import warnings
+
+from ...utils import split_and_load
+from ....metric import Loss as metric_loss
+from .... import autograd
+
+__all__ = ['EstimatorModel']
+
+class EstimatorModel(object):
+    """EstimatorModel Class for plug and play fit_batch & evaluate_batch
+
+    :py:class:`EstimatorModel` can be used to replace fit_batch() and evaluate_batch()
```

Comment:
You may want to provide more documentation here?

Reply:
OK, I'll add more explanation later.